aeron-driver/src/main/c/aeron_driver_conductor.c

```diff
@@ -726,14 +729,18 @@ bool aeron_ipc_publication_entry_has_reached_end_of_life(
 
 void aeron_ipc_publication_entry_delete(aeron_driver_conductor_t *conductor, aeron_ipc_publication_entry_t *entry)
 {
+    aeron_ipc_publication_t *publication = entry->publication;
+    conductor->context->remove_publication_cleanup_func(
+        publication->session_id, publication->stream_id, publication->channel_length, publication->channel);
+
     for (size_t i = 0, size = conductor->ipc_subscriptions.length; i < size; i++)
     {
         aeron_subscription_link_t *link = &conductor->ipc_subscriptions.array[i];
-        aeron_driver_conductor_unlink_subscribable(link, &entry->publication->conductor_fields.subscribable);
+        aeron_driver_conductor_unlink_subscribable(link, &publication->conductor_fields.subscribable);
     }
 
-    aeron_ipc_publication_close(&conductor->counters_manager, entry->publication);
-    entry->publication = NULL;
+    aeron_ipc_publication_close(&conductor->counters_manager, publication);
+    publication = NULL;
```

Comment:
Are you nulling the correct pointer here?

Reply:
No, I'm not. Good catch. It was a result of extracting a variable.

Reply:
Fixed.